worlds/pokemon_rb/encounters.py

```diff
@@ -106,21 +106,21 @@ def process_pokemon_locations(self):
     placed_mons = {pokemon: 0 for pokemon in poke_data.pokemon_data.keys()}
 
     mons_list = [pokemon for pokemon in poke_data.first_stage_pokemon if pokemon not in poke_data.legendary_pokemon
-                 or self.multiworld.randomize_legendary_pokemon[self.player].value == 3]
-    if self.multiworld.randomize_legendary_pokemon[self.player] == "vanilla":
+                 or self.options.randomize_legendary_pokemon.value == 3]
+    if self.options.randomize_legendary_pokemon == "vanilla":
         for slot in legendary_slots:
             location = self.multiworld.get_location(slot.name, self.player)
             location.place_locked_item(self.create_item("Static " + slot.original_item))
-    elif self.multiworld.randomize_legendary_pokemon[self.player] == "shuffle":
-        self.multiworld.random.shuffle(legendary_mons)
+    elif self.options.randomize_legendary_pokemon == "shuffle":
```

Comment:
Then why even have string comparison in the Choice class?

Reply:
Core's stance is text comparisons are perfectly fine, as they throw a loud error when they are faulty.

Constant comparisons (what agilbert1412 is suggesting) are slightly cleaner but take up more space

agilbert1412 has good reasons to suggest the constant comparisons, but feel free to keep the text comparisons

Reply:
As with most of my suggestions that aren't straight up "this is a bug", I always consider it your choice to follow it or not. I say what I would do, and if desired, I'll elaborate on it, and you make the final call. I'd never actively block a PR for a preference thing like that.

But since you did ask, here's why using the predefined value instead of a string literal is better, in my opinion:

If you ever rename an option or change its underlying value (ex: Inserting another value in a `Choice` and bumping several numbers), if you are using the constants, you'll be fine, but if you are using literals and magic numbers, you will need to track them all down and change them to avoid bugs. It is not particularly difficult to track them down, just potentially tedious, but the real risk is *forgetting to do it*­.
Another thing is potential typos. If you typo the variable, Python will crash loudly, if you ever even run it (you can also notice a squiggly line), but if you typo in a string literal, your condition will simply silently be false forever. Which makes it much harder to notice.